internal/repository/skr/webhook/skrwebhook_repo.go

```diff
@@ -0,0 +1,166 @@
+	resources = append(resources,
+		ResourceRef{
+			Name:       skrwebhookresources.SkrResourceName,
+			APIVersion: admissionregistrationv1.SchemeGroupVersion.String(),
+			Kind:       "ValidatingWebhookConfiguration",
+		},
+		ResourceRef{
+			Name:       skrwebhookresources.SkrTLSName,
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+	)
+
+	return &ResourceRepository{
+		resources:           resources,
+		skrClientCache:      skrClientCache,
+		remoteSyncNamespace: remoteSyncNamespace,
+	}
+}
+
+// ResourcesExist checks if any of the skr-webhook resources exist in the SKR cluster for the given Kyma.
+func (r *ResourceRepository) ResourcesExist(kymaName string) (bool, error) {
+	kymaNamespacedName := types.NamespacedName{
+		Name:      kymaName,
+		Namespace: r.remoteSyncNamespace,
+	}
+
+	skrClient := r.skrClientCache.Get(kymaNamespacedName)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	errGrp, grpCtx := errgroup.WithContext(ctx)
+	resourceExists := make(chan bool, 1)
+
+	for idx := range r.resources {
+		resIdx := idx
+		errGrp.Go(func() error {
+			select {
+			case <-grpCtx.Done():
+				// Short-circuit if context is cancelled (resource already found)
+				return nil
+			default:
+			}
+
+			ref := r.resources[resIdx]
+			resource := &unstructured.Unstructured{}
+			resource.SetGroupVersionKind(schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind))
+			resource.SetName(ref.Name)
+			resource.SetNamespace(r.remoteSyncNamespace)
+
+			err := skrClient.Get(grpCtx, client.ObjectKeyFromObject(resource), resource)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// Resource does not exist, continue checking others
+					return nil
+				}
+				return fmt.Errorf("failed to check resource %s: %w", ref.Name, err)
+			}
+			select {
+			case resourceExists <- true:
+				cancel()
+			default:
+			}
+			return nil
+		})
+	}
+
+	if err := errGrp.Wait(); err != nil {
+		return false, err
+	}
+
+	select {
+	case <-resourceExists:
+		return true, nil
+	default:
+		return false, nil
+	}
+}
+
+// DeleteWebhookResources deletes all skr-webhook resources from the SKR cluster for the given Kyma.
+func (r *ResourceRepository) DeleteWebhookResources(ctx context.Context, kymaName string) error {
+	kymaNamespacedName := types.NamespacedName{
+		Name:      kymaName,
+		Namespace: r.remoteSyncNamespace,
+	}
+
+	skrClient := r.skrClientCache.Get(kymaNamespacedName)
+
+	errGrp, grpCtx := errgroup.WithContext(ctx)
+
+	for idx := range r.resources {
+		resIdx := idx
+		errGrp.Go(func() error {
+			ref := r.resources[resIdx]
+			resource := &unstructured.Unstructured{}
+			resource.SetGroupVersionKind(schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind))
+			resource.SetName(ref.Name)
+			resource.SetNamespace(r.remoteSyncNamespace)
+
+			err := skrClient.Delete(grpCtx, resource)
+			if err != nil {
+				return fmt.Errorf("failed to delete resource %s: %w", ref.Name, err)
+			}
```

Comment:
I think we already need to ignore not found here. If I understant the errGrp.Wait correclty, it returns the first non nill error. If we have something like:

- err1: not found
- err2: some real error

Then the `err` below is `err1` and since we ignore not found there, we would miss the real error in err2.